ow_core/developer_tools.php

```diff
@@ -0,0 +1,383 @@
+                    }
+                    else if ( strstr($key, '_CTRL_') )
+                    {
+                        $rndArray['ctrl'] = $addItem;
+                    }
+                    else
+                    {
+                        $rndArray['cmp'][] = $addItem;
+                    }
+                }
+
+                $viewRenderer->assignVar('renderedItems', array('items' => $rndArray, 'count' => ( count(OW_Renderable::getRenderedClasses()) - 2 )));
+
+                $queryLog = OW::getDbo()->getQueryLog();
+                foreach ( $queryLog as $key => $query )
+                {
+                    if ( isset($_GET['pr_query_log_filter']) && strlen($_GET['pr_query_log_filter']) > 3 )
+                    {
+                        if ( !strstr($query['query'], $_GET['pr_query_log_filter']) )
+                        {
+                            unset($queryLog[$key]);
+                            continue;
+                        }
+                    }
+
+                    if ( isset($query['params']) && is_array($query['params']) )
+                    {
+                        $queryLog[$key]['params'] = var_export($query['params'], true);
+                    }
+                }
+
+                $viewRenderer->assignVar('database', array('qet' => OW::getDbo()->getTotalQueryExecTime(), 'ql' => $queryLog, 'qc' => count($queryLog)));
+
+                //events
+                $eventsData = OW::getEventManager()->getLog();
+                $eventsDataToAssign = array('bind' => array(), 'calls' => array());
+
+                foreach ( $eventsData['bind'] as $eventName => $listeners )
+                {
+                    $listenersList = array();
+
+                    foreach ( $listeners as $priority )
+                    {
+                        foreach ( $priority as $listener )
+                        {
+                            if ( is_array($listener) )
+                            {
+                                if ( is_object($listener[0]) )
+                                {
+                                    $listener = get_class($listener[0]) . ' -> ' . $listener[1];
+                                }
+                                else
+                                {
+                                    $listener = $listener[0] . ' :: ' . $listener[1];
+                                }
+                            }
+                            else if ( is_string($listener) )
+                            {
+                                
+                            }
+                            else
+                            {
+                                $listener = 'ClosureObject';
+                            }
+
+                            $listenersList[] = $listener;
+                        }
+                    }
+
+                    $eventsDataToAssign['bind'][] = array('name' => $eventName, 'listeners' => $listenersList);
+                }
+

```

Comment:
Can you provide us with a little explanation of this code. Just add a few comments or add a few protected methods instead of this big function